lib/include/tick/array/basearray2d.h

```diff
@@ -132,6 +135,18 @@ class BaseArray2d : public AbstractArray1d2d<T> {
   // This method is defined in sparsearray2d.h
   Array2d<T> as_array2d();
 
+  bool compare(const BaseArray2d<T> &that) {
+    bool b = (this->_n_rows == that._n_rows) && (this->_n_cols == that._n_cols);
```

Comment:
I would find cleaner to first initialize `b` to `AbstractArray1d2d<T>::compare(that);`, so that we can rely on the fact that we will test next attributes only if the basic ones are already checked.
(and we can change `b` variable name :) )